types/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+pub use crate::data_types::{Column, DataTypeNode};
+use crate::decoders::{ensure_size, read_string};
+use crate::error::TypesError;
+pub use crate::leb128::put_leb128;
+pub use crate::leb128::read_leb128;
+use bytes::BufMut;
+
+pub mod data_types;
+pub mod decoders;
+pub mod error;
+pub mod leb128;
+
+pub fn parse_rbwnat_columns_header(buffer: &mut &[u8]) -> Result<Vec<Column>, TypesError> {
```

Comment:
Let's start enforcing documentation for new crates at least (`missing_docs = "warn"`). Otherwise, we will be forced to add a lot of unrelated documentation in one go in the future.

https://github.com/ClickHouse/clickhouse-rs/pull/160